test/test_glue.py

```diff
@@ -98,18 +99,60 @@ def test_list(memgraph_connection):
 
     cursor.execute("RETURN $value",
                    {'value': [1, 2, None, True, False, 'abc', []]})
-    assert cursor.fetchall() == [([1, 2, None, True, False, 'abc', []], )]
+    result = cursor.fetchall()
+    assert result == [([1, 2, None, True, False, 'abc', []], )]
+    value_from_result = result[0][0][6]
```

Comment:
Is it worthwhile checking the refcount of `result[0][0]` (the list that contains all the results), or is it handled the same as the `result[0][0][6]`?

Reply:
So the `result` is a list of rows and its type is list of tuples. In this case the tuple has only one item, which is a list. So checking the refs of `result[0][0]` means checking the refs of the tuple elements. That is done in [`test_tuple`](https://github.com/memgraph/pymgclient/pull/13/files/27dd2bc3f6b1d705e1ca38b9dbee8ac1ebfeec6a#diff-2de5e8f77e139b1b94f0dd90b765370c24f358d82f8d6624ebce07120b40c0d9R212) I think. Though checking the refs of `result` and `result[0]` is not done at all. Considering that one of the memory leak was caused by the latter one, I will add tests for it also.

Reply:
Oh yeah, I see now that it's tested in the `test_tuple`.
>Though checking the refs of result and result[0] is not done at all. Considering that one of the memory leak was caused by the latter one, I will add tests for it also.

:+1:

Reply:
The new test discovered another leak, now in `cursor_fetchall`. I double-checked all `PyList_*`, `PyDict_*` and `PyTuple_*` function/macro calls, but couldn't spot anything problematic.